src/plugins/intel_gpu/src/graph/reshape.cpp

```diff
@@ -60,7 +60,14 @@ std::vector<layout> reshape_inst::calc_output_layouts(reshape_node const& /*node
     // we return output_partial_shape taken from the original model intead of something like PartialShape::dynamic(rank)
     // as ngraph may refine output shape using interval arithmetic
     if ((memory_deps.empty() && prim->output_pattern.empty()) || input_layout.is_dynamic()) {
-        return { layout{prim->output_partial_shape, input_layout.data_type, format::adjust_to_rank(input_layout.format, prim->output_partial_shape.size())} };
+        if (prim->output_partial_shape.size() > 0) {
```

Comment:
Why don't you use `prim->output_parital_shape.ranke().is_static()` ?

Reply:
rank can be static though the shape is dynamic. 
The meaning of "output_partial_shape" is set is that the predefined output shape is set and it is partial shape..

Reply:
From line 64, prim->output_partial_shape.size() is used so I checked the size is larger than 0.